src/zenml/stack/stack_component.py

```diff
@@ -242,6 +247,35 @@ def _is_part_of_active_stack(self) -> bool:
         # (see https://github.com/python/mypy/issues/13319).
         __getattribute__ = __custom_getattribute__
 
+    @validator("*", pre=True)
+    def _convert_json_strings(cls, value: Any, field: ModelField) -> Any:
+        """Converts potential JSON strings passed via the CLI.
+
+        Args:
+            value: The value to convert.
+
+        Returns:
+            The converted value.
+
+        Raises:
+            ValueError: If the value is an invalid JSON string.
+        """
+        if isinstance(value, str):
+            if get_origin(field.outer_type_) in [
```

Comment:
Just a small reminder, aside from the `get_origin`, there are a few more things that need to change here when we switch to pydantic v2 such as the `type_`, `outer_type_`, `parse_raw` and `dict()`